salt/docs/config/nginx.docs-redirects.conf

```diff
@@ -182,11 +182,21 @@ location ~ ^/(searchindex.js|objects.inv)$ {
 # Redirect ``$request_uri`` -> ``$request_uri.html``,
 # where the latter is a valid webpage.
 location ~ ^/((2|3)(\.[0-9]+)?|dev)/\w+/[\d\w\.]+(?!\.html)$ {
+    add_header Surrogate-Key en/$1 always;
     if (-f "${request_filename}.html") {
         return 301 https://$host:$request_uri.html;
     }
 }
 
+# Add the Surrogate-Key for bulk CDN purging
+# xref https://docs.fastly.com/en/guides/working-with-surrogate-keys
+location ~ ^/((2|3)(\.[0-9]+)?|dev)/ {
+    add_header Surrogate-Key en/$1 always;
+}
+location ~ ^/(es|fr|id|it|ja|ko|pl|pt-br|tr|uk|zh-cn|zh-tw)/((2|3)(\.[0-9]+)?|dev)/ {
+    add_header Surrogate-Key $1/$2 always;
+}
```

Comment:
Ideally I had hoped to use some sort of non-binding match construct to add the headers, but it seems that NGINX `location` directives will always bind. The end goal is simply to add some unique prefix for every second level path (and English) that we can use to purge. I think the approach here is fine given every other rule here is a redirect, but worth pointing out.

Fastly [has some text](https://www.fastly.com/documentation/guides/concepts/edge-state/cache/purging/#adding-surrogate-keys) about adding surrogate keys via `vcl_fetch`, but I don't know where I'd configure that.

A